### PR TITLE
Cisco ISE fixes for monitoring data purge and additional example for alarm

### PR DESCRIPTION
**Pull request by philippkahr**
Hi,

the ISE alarms can also be like this:
```
<182>Dec 11 10:13:45 abcdefghihklm002 CISE_Alarm INFO: Satellite Authorization Renewal success
```

which our default grok pattern:

```
  - grok:
      field: _tmp.message
      tag: grok_message_details
      if: ctx._tmp?.message != null
      patterns:
        - '^%{NOTCOLON:cisco_ise.log.log_severity_level}%{SPACE}: %{NOTCOLON:event.action}%{SPACE}: %{GREEDYDATA:message}$'
        - '^%{DATA:cisco_ise.log.message.id} %{DATA:cisco_ise.log.segment.total:long} %{DATA:cisco_ise.log.segment.number:long} %{GREEDYDATA:message}$'
```

cannot handle properly, that is why I introduced an on_failure processor to deal with a bit weirder kinds of grok patterns. I am not too much of a fan of doing an grok in the `on_failure` as in: if those don't match then this should match, but looking at it, it's the only way I can ensure that nothing of the existing is broken or altered and it is just a failsafe. Additionally, this should be helpful in regards to the logs+ project. Yes, the grok might fail, but still some renames and whatnot can work instead of stopping completely.

I also added a new type of log and added an example for it.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
ted metrics or logs

### CODE REVIEW

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_alarm.yml
Comment (efd6) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_alarm.yml:1]: Do we need the changes in this file? A don't see anything beyond whitespace and the final on_error addition.

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -30,9 +30,18 @@ processors:
...
-            field: error.message
-            value: '{{{ _ingest.on_failure_processor_tag }}}: {{{_ingest.on_failure_message}}}'
+        - grok:
+            description: "Events like: 'INFO: Satellite Authorization Renewal success'"
+            field: _tmp.message
+            tag: grok_message_details_fallback
+            if: ctx._tmp?.message != null
+            patterns:
+                - '^%{WORD:cisco_ise.log.log_severity_level}: %{GREEDYDATA:message}'
+                - '^%{GREEDYDATA:message}'
```
Comment (efd6) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/default.yml:40]: Can this not be done with additional patterns in the parent?

File: packages/cisco_ise/data_stream/log/_dev/test/pipeline/test-pipeline-cise-alarm.log [hunk 2]
```diff
@@ -21,3 +21,4 @@
 <178>Sep 26 02:15:47 abcdefghihklm002 CISE_Alarm WARN: Profiler SNMP Request Failure : Server= abcdefghihklm003; NAD Address=10.111.11.111; Error Message=Request timed out.
 <179>Sep 26 02:15:47 abcdefghihklm002 CISE_Alarm INFO: EAP Connection Timeout : Server=aaaaaaaa05nm001; NAS IP Address=216.160.83.56; NAS Identifier=N/A
 <178>Sep 24 16:03:27 abcdefghihklm002 CISE_Alarm WARN: Dynamic Authorization Failed for Device : Server=safsfasdsacdefg001; Calling Station Id=C0:25:AA:AA:AA:CA; Network device IP=89.160.20.128; Network Device Name=AAAA-1111-1; Failure Reason=11213 No response received from Network Access Device after sending a Dynamic Authorization request
+<182>Dec 11 10:13:45 abcdefghihklm002 CISE_Alarm INFO: Satellite Authorization Renewal success
```
Comment (efd6) [inline packages/cisco_ise/data_stream/log/_dev/test/pipeline/test-pipeline-cise-alarm.log:24]: Final new line

File: packages/cisco_ise/data_stream/log/_dev/test/pipeline/test-pipeline-monitoring-data-purge-audit.log [hunk 3]
```diff
@@ -0,0 +1 @@
+<182>Dec 12 05:11:05 cisco-ise-host CISE_MONITORING_DATA_PURGE_AUDIT 2023-12-12 04:16:26.526 +0100 60198 INFO null: MnT purge event occurred, MESSAGE=Total Data threshold_space = 265 GB, used_space = 5 GB
```
Comment (efd6) [inline packages/cisco_ise/data_stream/log/_dev/test/pipeline/test-pipeline-monitoring-data-purge-audit.log:1]: New line

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_monitoring_data_purge_audit.yml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    - "^%{TIMESTAMP_ISO8601:_tmp.timestamp} %{ISO8601_TIMEZONE:event.timezone} %{DATA:cisco_ise.log.message.code}\
+      \ %{DATA:log.syslog.severity.name} %{DATA:cisco_ise.log.message.description},\
+      \ %{GREEDYDATA:cisco_ise.log.log_details_raw},"
+- grok:
+    field: "cisco_ise.log.message.description"
+    if: "ctx.cisco_ise?.log?.message?.description != null && ctx.cisco_ise.log.message.description\
+      \ != ''"
+    patterns:
+    - "^%{DATA:event.action}:"
+    ignore_failure: true
```
Comment (efd6) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_monitoring_data_purge_audit.yml:18]: Can we do this with a dissect?

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_monitoring_data_purge_audit.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
+---
+processors:
+- set:
```
Comment (efd6) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_monitoring_data_purge_audit.yml:3]: Match indentation with the other pipelines in the package.

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_monitoring_data_purge_audit.yml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    field: "cisco_ise.log.log_details.MESSAGE"
+    target_field: "cisco_ise.log.log_details.message"
+    ignore_missing: true
+on_failure:
+  - set:
+      field: event.kind
+      value: pipeline_error
+  - append:
+      field: error.message
+      value: '{{{ _ingest.on_failure_message }}}'
```
Comment (efd6) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_monitoring_data_purge_audit.yml:101]: Final new line

**Comment by mrodm** [comment]
Hi @philippkahr, please update your branch with the latest contents from main branch. There was an important PR merged updating the CI pipelines. Thanks!

**Pull request closed without merging**